src/substrate/filters/enricher.ts

```diff
@@ -198,7 +255,7 @@ export async function Enrich(
       case EventKind.Bonded:
       case EventKind.Unbonded: {
         const [ stash, amount ] = event.data as unknown as [ AccountId, Balance ] & Codec;
-        const controllerOpt = await api.query.staking.bonded<Option<AccountId>>(stash);
+        const controllerOpt = await api.query.staking.bonded.at<Option<AccountId>>(hash,stash);
```

Comment:
In theory we should change every call to use `.at` if we were aiming to make Chain Events fully historical.